### PR TITLE
[Feature]:  Added Wishlist Page and functionality

### PR DESCRIPTION
**Pull request by munjalSaksham71**
Description: Added the wishlist Page and the functionality of it.

If we are click on ```Move to cart``` then the product will go to cart and remove from Wishlist page.

### CODE REVIEW

**Review by Kothalkarpriya — APPROVED** [review]
Looks good to me

File: src/components/Wishlist/WishListing.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import './Wishlist.css'
+
+const WishListing = () => {
+    const { wishlistState: {wishlist}, wishlistDispatch } = useWishlist();
+    const { cartDispatch } = useCart();
+    
+    const moveToCartHandler = (prod) => {
+        cartDispatch({type: 'ADD_TO_CART', payload: prod});
+        wishlistDispatch({type: 'REMOVE_FROM_WISHLIST', payload: prod})
+    }
```
Comment (mariyasada) [inline src/components/Wishlist/WishListing.jsx]: use product instead of prod

File: src/context/wishlist-context.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+const WishlistContext = createContext();
+
+const WishlistContextProvider = ({children}) => {
+    const wishlistReducer = (state, action) => {
+        switch (action.type) {
+            case 'ADD_TO_WISHLIST':
+                return {...state, wishlist: [...state.wishlist, action.payload]};
+            case 'REMOVE_FROM_WISHLIST':
+                return {...state, wishlist: state.wishlist.filter((w) => w._id !== action.payload._id) };
+            default:
```
Comment (mariyasada) [inline src/context/wishlist-context.js]: use proper name instead of w

**Review by mariyasada — APPROVED** [review]
look at the comments

**Pull request merged**